src/app/activity/user/RegisterWallet.ts

```diff
@@ -0,0 +1,35 @@
+import { Db, UpdateWriteOpResult } from "mongodb";
+import { UpsertUserWalletRequest } from "../../requests/UpsertUserWalletRequest";
+import { UserCollection } from "../../types/user/UserCollection";
+import MongoDbUtils from "../../utils/MongoDbUtils";
+import Log from "../../utils/Log";
+
+export const upsertUserWallet = async (request: UpsertUserWalletRequest): Promise<any> => {
+    return await dbHandler(request);
+}
+
+const dbHandler = async (request: UpsertUserWalletRequest): Promise<void> => {
+    const db: Db = await MongoDbUtils.connect('bountyboard');
+    const userCollection = db.collection('user');
+
+	const dbUserResult: UserCollection = await userCollection.findOne({
```

Comment:
I think you can use findAndModify as one operation here: https://stackoverflow.com/questions/16358857/mongodb-atomic-findorcreate-findone-insert-if-nonexistent-but-do-not-update